service/controller/resource/ipam/spec.go

```diff
@@ -15,7 +15,19 @@ type Checker interface {
 // Collector implementation must return all networks that are allocated on any
 // given moment. Failing to do that will result in overlapping allocations.
 type Collector interface {
-	Collect(ctx context.Context) ([]net.IPNet, error)
+	Collect(ctx context.Context, obj interface{}) ([]net.IPNet, error)
```

Comment:
Added `obj interface{}`, since we need to pass reconciled CR for node pools implementation. Existing `AzureConfig` implementation will just ignore this additional parameter.